### PR TITLE
feat(route): 金十数据 主题文章

### LINKED ISSUE #12666: 金十数据 主题文章
### 类型

金融

### 网站地址

https://xnews.jin10.com/topic/396

### 网站描述

汇聚金融投资行业的各类数据和资讯。

### 需要生成什么内容？

主题文章

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by miles170**
## 该 PR 相关 Issue / Involved Issue

Close #12666

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/jin10/topic/396
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/jin10/topic.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+
+module.exports = async (ctx) => {
+    const currentUrl = `https://xnews.jin10.com/topic/${ctx.params.id}`;
+    const response = await got(currentUrl);
+    const $ = cheerio.load(response.data);
```
Comment (TonyRL) [inline lib/v2/jin10/topic.js:7]: API is available: `https://reference-api.jin10.com/topic/getById?id=${id}`

Comment (miles170) [reply lib/v2/jin10/topic.js:7]: https://reference-api.jin10.com/topic/getById?id=396 returns `HTTP ERROR 502`

File: lib/v2/jin10/topic.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const items = await Promise.all(
+        $('.jin10-news-list-item-info a')
+            .toArray()
+            .map((a) => {
+                const link = $(a).attr('href');
+
+                return ctx.cache.tryGet(link, async () => {
+                    const response = await got(link);
+                    const $ = cheerio.load(response.data);
```
[the remark this answers is not in the document]
Comment (miles170) [reply lib/v2/jin10/topic.js:16]: https://reference-api.jin10.com/reference/getOne?id=114529&type=news returns `HTTP ERROR 502`

File: lib/v2/jin10/topic.js (hunk 1 above)
Comment (miles170) [reply lib/v2/jin10/topic.js:7]: https://reference-api.jin10.com/topic/getById?id=396 returns `HTTP ERROR 502`.
Do I need to set some secret keys? I couldn't find the docs.

Comment (TonyRL) [reply lib/v2/jin10/topic.js:7]: You need to set `x-app-id` and `x-version` header.

Comment (miles170) [reply lib/v2/jin10/topic.js:7]: I couldn't find what value to set for `x-app-id`. May I ask where you found that the API is called?

Comment (TonyRL) [reply lib/v2/jin10/topic.js:7]: https://www.gcores.com/collections, F12 and observe

Comment (miles170) [reply lib/v2/jin10/topic.js:7]: Never mind, I found it here: https://github.com/bafenzhiyimiao/foreign_dart_9/blob/fd0f4809ec78d00f5906a8647727818ef82deaa0/lib/choice/visual_screen.dart#L74﻿

**Pull request merged**